README.md

```diff
@@ -93,82 +93,182 @@ EncryptionParamaters = {
+dictionary IdentityProviderConfig {
+  required string scheme;
 
-    CredentialStorageDuration desiredStorageDuration;  // Not providing this is equivalent to not asking to store.
+  // Common fields across all identity provider types:
+  required DOMString nonce;
 };
 
-dictionary CredentialDocument {
-    object data;  // The CBOR encoded `CredentialDocument` defined above.
+// extends the IdentityProviderConfig with mDoc-specific attributes
+partial dictionary IdentityProviderConfig  {
+  // For use where `scheme` is `mdoc`:
+
+  // base64-encoded reader HPKE public key.
+  required DOMString readerPublicKey;
+
+  // base64-encoded reader certification chain, if the platform requires it.
+  optional sequence<DOMString> certificates;
+
+  required DOMString documentType;
+  required sequence<MdocRequestedElement> requestedElements;
 };
 
-dictionary RequestConfiguration {
-    required DOMString nonce;
+dictionary MdocElement {
+  required DOMString namespace;  // As defined in ISO 18013-5 clause 8.
+  required DOMString name;
 };
 
-[
-    SecureContext,
-    Exposed=Window,
-] interface CredentialRequest {
-    constructor(DOMString requesterIdentity, CredentialDocumentDescriptor documentDescriptor);  // This throws if anything in the `documentDescriptor` is not recognized (e.g. an invalid `documentType`).
+dictionary MdocRequestedElement : MdocElement{
+  boolean critical = true;
+  DOMString oneOfGroup;
+};
 
-    Promise<CredentialDocument> requestDocument(RequestConfiguration configuration);
 
-    Promise<undefined> abort();
+partial dictionary IdentityProviderConfig {
+  // ...
+  // Federated provider specific configs, not covered here.
+  // ...
 };
 ```
 
+There is always a bijection from `IdentityCredentialRequestOptions` to JSON strings, and that bijection is implemented by `JSON.parse` and `JSON.stringify`. This allows requests to be generated on the server and easily transported to the user agent.
+
+When an `IdentityCredentialRequestOptions` is used with [`get`](https://www.w3.org/TR/credential-management-1/#dom-credentialscontainer-get), the result is an `IdentityCredential`:
+
+```webidl
+[Exposed=Window, SecureContext]
+interface IdentityCredential : Credential {
+  // The CBOR encoded `CredentialDocument` defined above, wrapped in base64.
+  DOMString token;
+};
+```
+
+The `id` of the `IdentityCredential` for an mdoc response would be `mdoc`.
 
 ## Examples
 
+Requesting specific attributes from a driver’s license:
+
 ```js
-// Driver's License
-let mDLCredentialRequest = new CredentialRequest(certificate, {
-    documentType: "org.iso.18013.5.1.mDL",
-    requestedElements: [
+let request = {
+  identity: {
+    retention: {
+      days: 90,
+    },
+    providers: [{
+      scheme: “mdoc”,
+      nonce: "gf69kepV+m5tGxMyMF0cnn9NCnRHez/LUIsFtLi6pwg=",
+      documentType: "org.iso.18013.5.1.mDL",
+      readerPublicKey: "ftl+VEHPB17r2oi6it3ENaqhOOB0AZbAkb5f4VlCPakpdNioc9QZ7X/6w...",
```

Comment:
Most of these keys are new (`documentType` excepted). Do these changes have to do with the subject of this PR, "Criticality"? If so, how?

In general, this PR has almost no description to it. Can we please add a PR description talking about what changes there are here?